src/server/player_sao.cpp

```diff
@@ -407,6 +408,11 @@ void PlayerSAO::setPlayerYaw(const float yaw)
 	UnitSAO::setRotation(rotation);
 }
 
+const std::string& PlayerSAO::getGUID()
+{
+	return m_player_name;
```

Comment:
Why not `m_player->getName()`?

Reply:
`m_player` lives shorter than the name (if you just replace this with `m_player->getName()`, you will get null pointer derefs in valid usage iirc).

there's also the question of what to do with expired player refs - we can still know their names and thus GUIDs.